lib/proposed/character_factory.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './character'
+require_relative './character_validator'
```

Comment:
Próbowałeś użyć buildera, w przypadku tego "zadania", bardziej sprawdzi się wzorzec `Factory`. `Builder` używa się, jeśli budowany obiekt ma różne możliwości konfiguracji i nie musi mieć wypełnionych wszystkich atrybutów. Tutaj lepiej wykorzystać Factory, które utworzy nam po prostu character na podstawie parametrów, i pozwala to rozszerzyć, gdyby dodać inny model. Builder w tych scenariuszach nie ma żadnego zastosowania a robi niepotrzebną dodatkową warstwę abstrakcji.